src/die_init.cpp

```diff
@@ -332,17 +341,33 @@ namespace Die
                 // Start advertising!
                 Stack::startAdvertising();
 
+                // Initialize common logic
+                initMainLogic();
+
             #if defined(DEBUG_FIRMWARE)
-                initDebugLogic();
+                if (!inValidation) 
+                {
+                    // Initialize debug logic
+                    initDebugLogic();
+                }
                 NRF_LOG_INFO("---------------");
             #else
-                // Initialize main logic manager
-                initMainLogic();
+                // Initialize main die logic
+                if (!inValidation) 
```

Comment:
The `if (!inValidation) { initDebugLogic(); }` is duplicated, move block out of `#if defined(DEBUG_FIRMWARE) #else` and add comment to explain why we don't call it in validation mode